apollo-runtime/src/commonMain/kotlin/com/apollographql/apollo3/internal/MultipartReader.kt

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright (C) 2020 Square, Inc.
+ * Copyright (c) 2022 Apollo Graph, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.apollographql.apollo3.internal
+
+import com.apollographql.apollo3.api.http.HttpHeader
+import com.apollographql.apollo3.exception.ApolloException
+import okio.Buffer
+import okio.BufferedSource
+import okio.ByteString.Companion.encodeUtf8
+import okio.Closeable
+import okio.Options
+import okio.Source
+import okio.buffer
+
+/**
+ * This was copied from OkHttp and slightly modified to work in a Multiplatform project.
+ *
+ * Original source: https://github.com/square/okhttp/blob/parent-4.9.2/okhttp/src/main/kotlin/okhttp3/MultipartReader.kt
```

Comment:
Heads up we have another multipart reader [there](https://github.com/apollographql/apollo-kotlin/blob/f476e024c0c74e222270cc357ee48202df43ee0a/tests/integration-tests/src/commonTest/kotlin/test/FileUploadTest.kt#L155). Might be worth removing it?